src/threadpool_scheduler.cpp

```diff
@@ -64,15 +69,77 @@ struct threadpool_data {
 	std::condition_variable shutdown_complete_event;
 #endif
 };
-
+    
+#if defined(EMULATE_PTHREAD_THREAD_LOCAL)
+static pthread_key_t owning_threadpool_key;
+static pthread_key_t thread_id_key;
+    
+struct pthread_emulation_threadpool_data_initializer {
+    pthread_emulation_threadpool_data_initializer()
+	{
+        pthread_key_create(&owning_threadpool_key, nullptr);
+        pthread_key_create(&thread_id_key, nullptr);
+    }
+        
+    ~pthread_emulation_threadpool_data_initializer()
+    {
+        pthread_key_delete(owning_threadpool_key);
+        pthread_key_delete(thread_id_key);
+	}
+};
+    
+static pthread_emulation_threadpool_data_initializer threadpool_data_key_initializer;
+    
+#else
 // Thread pool this thread belongs to, or null if not in pool
 static THREAD_LOCAL threadpool_data* owning_threadpool = nullptr;
 
 // Current thread's index in the pool
 static THREAD_LOCAL std::size_t thread_id;
+#endif
+    
+static void create_threadpool_data(threadpool_data* _owning_threadpool, std::size_t _thread_id)
```

Comment:
Avoid identifiers that start with an underscore, those are reserved for the standard library. Instead put the underscore at the end.